seatunnel-connectors-v2/connector-cassandra/src/main/java/org/apache/seatunnel/connectors/seatunnel/cassandra/sink/CassandraSinkWriter.java

```diff
@@ -0,0 +1,148 @@
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.cassandra.sink;
+
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.common.utils.ExceptionUtils;
+import org.apache.seatunnel.connectors.seatunnel.cassandra.client.CassandraClient;
+import org.apache.seatunnel.connectors.seatunnel.cassandra.config.CassandraConfig;
+import org.apache.seatunnel.connectors.seatunnel.cassandra.util.TypeConvertUtil;
+import org.apache.seatunnel.connectors.seatunnel.common.sink.AbstractSinkWriter;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.AsyncResultSet;
+import com.datastax.oss.driver.api.core.cql.BatchStatement;
+import com.datastax.oss.driver.api.core.cql.BoundStatement;
+import com.datastax.oss.driver.api.core.cql.ColumnDefinitions;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.core.type.DataType;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Slf4j
+public class CassandraSinkWriter extends AbstractSinkWriter<SeaTunnelRow, Void> {
+
+    private final CassandraConfig cassandraConfig;
+    private final SeaTunnelRowType seaTunnelRowType;
+    private final ColumnDefinitions tableSchema;
+    private final CqlSession session;
+    private BatchStatement batchStatement;
+    private List<BoundStatement> boundStatementList;
+    private List<CompletionStage<AsyncResultSet>> completionStages;
+    private final PreparedStatement preparedStatement;
+    private final AtomicInteger counter = new AtomicInteger(0);
+
+    public CassandraSinkWriter(CassandraConfig cassandraConfig, SeaTunnelRowType seaTunnelRowType, ColumnDefinitions tableSchema) {
+        this.cassandraConfig = cassandraConfig;
+        this.seaTunnelRowType = seaTunnelRowType;
+        this.tableSchema = tableSchema;
+        this.session = CassandraClient.getCqlSessionBuilder(
+            cassandraConfig.getHost(),
+            cassandraConfig.getKeyspace(),
+            cassandraConfig.getUsername(),
+            cassandraConfig.getPassword(),
+            cassandraConfig.getDatacenter()).build();
+        this.batchStatement = BatchStatement.builder(cassandraConfig.getBatchType()).build();
+        this.boundStatementList = new ArrayList<>();
+        this.completionStages = new ArrayList<>();
+        this.preparedStatement = session.prepare(initPrepareCQL());
+    }
+
+    @Override
+    public void write(SeaTunnelRow row) throws IOException {
+        BoundStatement boundStatement = this.preparedStatement.bind();
+        addIntoBatch(row, boundStatement);
+        if (counter.getAndIncrement() >= cassandraConfig.getBatchSize()) {
```

Comment:
Insufficient amount may always be kept in memory?

Reply:
Insufficient amount will be flush before the session close.

Reply:
May increase the probability of missing data, but can do this first